Cybozu Frontend Weekly 2022年6月まとめ (#94)
cybozu/frontend-expert: data/posts/frontend-weekly-2022-06.md

---
title: "Cybozu Frontend Weekly 2022年6月まとめ"
author: "nus3"
createdAt: "2022-07-25"
summary: "6月にFrontend Weeklyで話したフロントエンドの話題まとめ"
tags:
  - Cybozu
  - FrontendExpert
  - FrontendWeekly
---

> フロントエンドエキスパートチームでは、毎週火曜の 17 時に最近のフロントエンド情報を見てワイワイする Cybozu Frontend Weekly を開催しています。

Reviewer: > 毎週火曜の 17 時に

社内イベントがいつ開催してるかは（読んでる人からしたら）どうでもよさそうな情報なので `週に1回` でヨサソウかなと思いました

Author: > フロントエンドエキスパートチームでは週に 1 回、最近のフロントエンド情報を見てワイワイする Cybozu Frontend Weekly というイベントを社内で開催しています。

週に1回にしつつ、社内イベントっぽさを醸し出してみました

> 今回は Cybozu Frontend Weekly で 6 月に話した記事をご紹介します。

Reviewer: > 話した記事

話題に上がった記事 とかのほうがわかりやすそう

## 💬 Languages

### [Bringing forward the End-of-Life Date for Node.js 16 | Node.js](https://nodejs.org/en/blog/announcements/nodejs16-eol/)

Node.js v16 の EOL が 当初の予定より 7 ヶ月早まり 2023/9/11 になるとのことです。OpenSSL 1.1.1 側の EOL にあわせるのが理由だそうです。

## 📖 Framework, Library

### [RedwoodJS を Ruby on Rails と比較してみる](https://zenn.dev/mugi/articles/334f9556095a07)

> チームの[@mugi_uno](https://twitter.com/mugi_uno) さんが書いてくれた RedwoodJS と Ruby on Rails の比較記事。本人曰く、RedwoodRecord という ORM が Rails の ActiveRecord にかなり近い体験を提供してくれるので感動したらしいです。また、RedwoodJS は RedwoodJS を利用する企業に対してファンドを実施するなど市場拡大に対して精力的なようです。

Reviewer 2: 敬称なくても大丈夫です〜！

```suggestion
チームの[@mugi_uno](https://twitter.com/mugi_uno) が書いた RedwoodJS と Ruby on Rails の比較記事。本人曰く、RedwoodRecord という ORM が Rails の ActiveRecord にかなり近い体験を提供してくれるので感動したらしいです。また、RedwoodJS は RedwoodJS を利用する企業に対してファンドを実施するなど市場拡大に対して精力的なようです。
```

### [feat: create "@eslint-community GitHub organization" RFC #91](https://github.com/eslint/rfcs/pull/91)

eslint-community org を作って ESLint に関係あるパッケージのメンテナンスをするという提案が進行しているようです。

### [Headless UI v1.6, Tailwind UI team management, Tailwind Play improvements, and more](https://tailwindcss.com/blog/2022-05-23-headless-ui-v1-6-tailwind-ui-team-management)

Tailwind CSS のチームが最近行った活動のまとめです。

- HeadlessUI の 1.6 リリース
- VSCode の Tailwind CSS IntelliSense の拡張に Tailwind CSS language mode が追加
- Tailwind CSS templates が発表予定

### [Vue “vapor” - an experimental, Solid.js-inspired compilation strategy.](https://twitter.com/_jessicasachs/status/1532283507145420801)

> `vue/vapor` が Vue Amsterdam で発表されました。SolidJS をベースにしたコンパイルの仕組みとのことです。

Reviewer 2: マジでSolidJSで動くものみたいな印象受けちゃうので、

```suggestion
`vue/vapor` が Vue Amsterdam で発表されました。SolidJS の仕組みに着想を得た新しいコンパイルの仕組みとのことです。
```

とかのほうが語弊がなさそう

### [`lerna/lerna` v5.0.0 Release](https://github.com/lerna/lerna/releases/tag/v5.0.0)

> Lerna が Nx にが委譲されてから初のアップデートです。内部処理が Nx に切り替わリました。

Reviewer: ```suggestion
Lerna が Nx に委譲されてから初のアップデートです。内部処理が Nx に切り替わリました。
```

### [State of Vuenion](https://docs.google.com/presentation/d/1tPaimqwJEXinPYifwUvA4gLCI6SezZsViuJvieWGAug/edit)

Vue Amsterdam での Evan you の発表資料です。

> Vue 2.7 以降での更新内容や、直近リリース予定の Vue 3.3 での変更内容、エコシステムの状況などに加え、Vue の将来向けに検討されている新しいコンパイルの仕組みについてなどが紹介されています。

Reviewer: ```suggestion
Vue 2.7 の内容や、直近リリース予定の Vue 3.3 での変更内容、エコシステムの状況などに加え、Vue の将来向けに検討されている新しいコンパイルの仕組みについてなどが紹介されています。
```

2.7が2系の最終版なので

> ### [State of Vuenion](https://docs.google.com/presentation/d/1tPaimqwJEXinPYifwUvA4gLCI6SezZsViuJvieWGAug/edit)
> 
> Vue Amsterdam での Evan you の発表資料です。
> 
> Vue 2.7 以降での更新内容や、直近リリース予定の Vue 3.3 での変更内容、エコシステムの状況などに加え、Vue の将来向けに検討されている新しいコンパイルの仕組みについてなどが紹介されています。

Reviewer 2: こっちがあれば

https://github.com/cybozu/frontend-expert/pull/94/files#diff-4ca8b28906e804d2c993521074520d12dfaeabd77937fd070cbd18f9a08b3497R40-R42

のTweetの紹介は不要かも。

こちらのほうで 

```suggestion
### [State of Vuenion](https://docs.google.com/presentation/d/1tPaimqwJEXinPYifwUvA4gLCI6SezZsViuJvieWGAug/edit)

Vue Amsterdam での Evan you の発表資料です。

Vue 2.7 以降での更新内容や、直近リリース予定の Vue 3.3 での変更内容、エコシステムの状況などに加え、Vue の将来向けに検討されている vue/vapor と呼ばれる新しいコンパイルの仕組みについてなどが紹介されています。
```

とすればよさそう

### [Prettier 2.7](https://prettier.io/blog/2022/06/14/2.7.0.html)

https://twitter.com/PrettierCode/status/1536604535497854977

Prettier 2.7 がリリースされました。
TypeScript 4.7 サポートに加え、高速化のための --cache オプションが追加されたそうです。
チェック高速化に寄与するとのことなので、ぜひ試していきましょう。

### [Apple Music の新しいサイトで Svelte が採用された](https://twitter.com/kevmodrome/status/1534275183334277121?s=20&t=Fm2pmMNFKzMqzQTiUtLQSw)

https://bundlescanner.com/website/beta.music.apple.com%2Fus%2Fbrowse/all  
を見ると、Svelte が採用されているのがわかります。

### [Component Encyclopedia のアップデート](https://storybook.js.org/blog/component-encyclopedia/)

> さまざまなプロダクトの StoryBook コンポーネントを見れる Component Encyclopedia がアップデートされました。各コンポーネントの検索機能の追加などが追加されています。

Reviewer: ```suggestion
さまざまなプロダクトの Storybook コンポーネントを見れる Component Encyclopedia がアップデートされました。各コンポーネントの検索機能の追加などが追加されています。
```

- RFC 9230: Oblivious DNS over HTTPS

なお、RFC 公開を祝い HTTP RFC Publication Study という、
複数人の方で内容のキャッチアップを行うイベントが行われ、その動画が YouTube 上にアーカイブとして残っています。
https://www.youtube.com/watch?v=_hfG0HCufbs

### [Web 技術解体新書「第二章 Cache 解体新書」リリース](https://zenn.dev/jxck/books/cache-anatomia)

Web における Caching の概念を `Cache-Control` だけでなく関連するあらゆる仕様の側面から解説した Zenn book がリリースされました。

## 🎨 Design

### [The Surprising Truth About Pixels and Accessibility](https://www.joshwcomeau.com/css/surprising-truth-about-pixels-and-accessibility/)

px と rem/em どっち使う？という話。
フォントは rem 等を使って padding なんかは px でいいよね(場合による)、みたいな話でした。

px と rem のマッピングを良い感じにするために body のフォントサイズに 62.5% を設定するのは良くないよという話が印象的でした。
(1rem はあくまでユーザが読みやすい標準的なサイズにすべきという思想らしいです)

### [Roboto … But Make It Flex](https://material.io/blog/roboto-flex)

Google Fonts に Roboto Flex という新しい Variable Fonts が追加されました。

### [Building a button component](https://web.dev/building-a-button-component/)

ダークモード・ライトモードや a11y を意識したボタンコンポーネントの作り方の紹介です。`:is` を利用した `:hover`, `:focus` の一括設定や、 outline-offset を利用したフォーカス時のスタイル制御など、実際にすぐ現場で使えそうな例も多かったです。

### [Customizing Color Fonts on the Web](https://webkit.org/blog/12662/customizing-color-fonts-on-the-web/)

> @font-palette-values が Safari 15.4 以降でサポートされたカラーフォントに関しての記事です。カラーパレットの Override 方法や、実際に適用するときの指定方法などが紹介されています。

Reviewer: > @font-palette-values が Safari 15.4 以降でサポートされたカラーフォントに関しての記事です

なにが何なのか日本語的に分かりづらい感じがします

Author: 文章変えやした！

> Safari 15.4 から`@font-palette-values`がサポートされました。この記事では実際に`@font-palette-values`を使ってカラーフォントに定義されているパレットを指定する方法や、パレットの色を上書きする方法が紹介されています。

なお、RFC 公開を祝い HTTP RFC Publication Study という、
複数人の方で内容のキャッチアップを行うイベントが行われ、その動画が YouTube 上にアーカイブとして残っています。
https://www.youtube.com/watch?v=_hfG0HCufbs

### [Web 技術解体新書「第二章 Cache 解体新書」リリース](https://zenn.dev/jxck/books/cache-anatomia)

Web における Caching の概念を `Cache-Control` だけでなく関連するあらゆる仕様の側面から解説した Zenn book がリリースされました。

## 🎨 Design

### [The Surprising Truth About Pixels and Accessibility](https://www.joshwcomeau.com/css/surprising-truth-about-pixels-and-accessibility/)

px と rem/em どっち使う？という話。
フォントは rem 等を使って padding なんかは px でいいよね(場合による)、みたいな話でした。

px と rem のマッピングを良い感じにするために body のフォントサイズに 62.5% を設定するのは良くないよという話が印象的でした。
(1rem はあくまでユーザが読みやすい標準的なサイズにすべきという思想らしいです)

### [Roboto … But Make It Flex](https://material.io/blog/roboto-flex)

Google Fonts に Roboto Flex という新しい Variable Fonts が追加されました。

### [Building a button component](https://web.dev/building-a-button-component/)

ダークモード・ライトモードや a11y を意識したボタンコンポーネントの作り方の紹介です。`:is` を利用した `:hover`, `:focus` の一括設定や、 outline-offset を利用したフォーカス時のスタイル制御など、実際にすぐ現場で使えそうな例も多かったです。

### [Customizing Color Fonts on the Web](https://webkit.org/blog/12662/customizing-color-fonts-on-the-web/)

> @font-palette-values が Safari 15.4 以降でサポートされたカラーフォントに関しての記事です。カラーパレットの Override 方法や、実際に適用するときの指定方法などが紹介されています。

Reviewer 2: 記事の一番最後に、めっちゃ簡単なものでいいと思うので、
今回の記事全体通してのnus3の感想が入っているとフレンドリーな印象の記事になりそうだなって思いました！

Author: 締めの言葉追加したぜぃ！
https://github.com/cybozu/frontend-expert/pull/94/commits/94769d681ef7a4ab3d8af9bda11ddb9c818f4df8
